### PR TITLE
optimize ColumnString Append

### PR DESCRIPTION
**Pull request by 1261385937**
[https://github.com/ClickHouse/clickhouse-cpp/issues/197](url)

### CODE REVIEW

File: clickhouse/columns/string.cpp [hunk 1]
```diff
@@ -179,18 +137,21 @@ ColumnString::ColumnString(const std::vector<std::string> & data)
     }
 }
 
-ColumnString::~ColumnString()
-{}
+ColumnString::ColumnString(std::vector<std::string>&& data)
```
Comment (Enmk) [inline clickhouse/columns/string.cpp:140]: Please add a test for this constructor.

**Review by Enmk — CHANGES_REQUESTED** [review]
Thank you for your contribution! There are a couple of minor things to fix before this can be merged.

File: clickhouse/columns/string.cpp (hunk 1 above)
Comment (1261385937) [reply clickhouse/columns/string.cpp:140]: hi, ColumnsCase.StringInit has covered yet.

File: clickhouse/columns/string.cpp [hunk 2]
```diff
@@ -179,18 +137,21 @@ ColumnString::ColumnString(const std::vector<std::string> & data)
...
+    : Column(Type::CreateString())
```
[the remark this answers is not in the document]
Comment (1261385937) [reply clickhouse/columns/string.cpp:141]: excuse me ?

File: clickhouse/columns/string.h [hunk 3]
```diff
@@ -6,9 +6,17 @@
 #include <string_view>
 #include <utility>
 #include <vector>
+#include <algorithm>
+#include <type_traits>
```
Comment (Enmk) [inline clickhouse/columns/string.h:10]: Unnecessary includes

File: clickhouse/columns/string.cpp [hunk 4]
```diff
@@ -179,10 +178,22 @@ ColumnString::ColumnString(const std::vector<std::string> & data)
     }
 }
 
+ColumnString::ColumnString(std::vector<std::string>&& data)
+    : Column(Type::CreateString())
```
Comment (Enmk) [inline clickhouse/columns/string.cpp:182]:
```suggestion
    : ColumnString()
```

**Review by Enmk — CHANGES_REQUESTED** [review]
Only some minor changes left, everything else is Ok

**Comment by 1261385937** [comment]
Hi, Append(const char*) overload is necessary for gcc && clang

**Pull request merged**